### PR TITLE
feat: add "break-word" wrapping alongside "whitespace"

### PR DESCRIPTION
**Pull request by Ddystopia**
Code had generics initially, but it quickly became lifetime hell with GATs etc, so I made it into an enum. Match per line would be cheap, especially copared to other code.

### CODE REVIEW

File: src/view/text/break_word_wrap.rs [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+            }
+        }
+        None
+    }
+}
+
+impl<'a, F: FontMetrics + 'a> Iterator for BreakWordWrap<'a, F> {
+    type Item = &'a str;
+
+    fn next(&mut self) -> Option<Self::Item> {
```
[the remark this answers is not in the document]
Comment (Ddystopia) [reply src/view/text/break_word_wrap.rs]: To be honest, I don't fully understand the original code with whitespaces, like many tweaks to it don't break any tests so I don't know the motivation, thus code for break-word is trying to be as close as possible to not break any edge cases I am not aware of, that were discovered for whitespaces.

File: src/lib.rs [hunk 2]
```diff
@@ -1,7 +1,4 @@
-#![no_std]
-
-#[cfg(any(test, feature = "std"))]
-extern crate std;
```
Comment (riley-williams) [inline src/lib.rs]: This is intentional, see: https://www.reddit.com/r/rust/comments/1hs6spy/psa_for_std_feature_in_no_std_libraries/

File: src/view/text.rs [hunk 3]
```diff
@@ -141,10 +158,15 @@ where
...
-        let wrap = WhitespaceWrap::new(self.text.as_ref(), offer.width, &metrics);
         let mut size = Size::zero();
-        // TODO: actually calculate this
         let line_ranges = heapless::Vec::new();
+        let mut whitespace = WhitespaceWrap::new(self.text.as_ref(), offer.width, &metrics);
+        let mut word = BreakWordWrap::new(self.text.as_ref(), offer.width, &metrics);
+        let wrap = core::iter::from_fn(|| match self.wrap {
+            WrapStrategy::Whitespace => whitespace.next(),
+            WrapStrategy::BreakWord => word.next(),
+        });
```
[the remark this answers is not in the document]
Comment (riley-williams) [reply src/view/text.rs]: Nope nevermind this is super annoying, you're right 😆

**Review by riley-williams — DISMISSED** [review]
I believe all the issues caused by integrating with my precise bounds PR have been resolved.

**Comment by Ddystopia** [comment]
Oh I'm sorry I totally forgot that PR exists, I was pulled into other stuff at work 😅

I will test and review stuff tomorrow, okay?

**Comment by riley-williams** [comment]
Sure, there's no rush. Appreciate all your help.
